Automattic/pocket-casts-ios
# Settings Sync: Set Modified Date on settings import

| **Depends On** | https://github.com/Automattic/pocket-casts-ios/pull/1615 |
|-|-|

Set the Modified Date on all imported settings to 1 second after the epoch date.

* This ensures that imported settings are sent to the server (any settings without a date are not).
* By defaulting to 1 second after, the setting is stored by the server. A default epoch date is effectively a "null" value and will not replace the default value.

## To test

* Change the return value of `shouldEnableSyncedSettings` in `FeatureFlag.swift` to `false`
* Build and run
* Log in to an account which hasn't used synced settings
* Change a few settings
* Enable the `newSettingsStorage` and `settingsSync` feature flags
* Restart the app
* Pull to refresh or tap "Refresh Now" in profile
* Ensure changed settings remain
* Delete the app
* Change `shouldEnableSyncedSettings` in `FeatureFlag.swift` back to its original implementation or set to `true`
* Build and run
* Log in to the same account from above
* Verify that the changed settings are the same as when changed above

## Checklist
